scripts/scil_register_tractogram.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import argparse
+import os
+
+from dipy.align.streamlinear import whole_brain_slr
+from nibabel.streamlines.array_sequence import ArraySequence
+import numpy as np
+
+
+from scilpy.io.streamlines import ichunk, load_tractogram_with_reference
+from scilpy.io.utils import (add_overwrite_arg, add_reference, add_verbose_arg,
+                             assert_inputs_exist, assert_outputs_exist)
+
+DESCRIPTION = """
+Generate a linear transformation matrix from the registration of
+2 tractograms. Typically, this script is run before
+scil_apply_transform_to_tractogram.py.
+
+For more informations on how to use the various registration scripts
+see the doc/tractogram_registration.md readme file
+"""
+
+EPILOG = """
+References:
+[1] E. Garyfallidis, O. Ocegueda, D. Wassermann, M. Descoteaux
+Robust and efficient linear registration of white-matter fascicles in the
+space of streamlines, NeuroImage, Volume 117, 15 August 2015, Pages 124-140
+(http://www.sciencedirect.com/science/article/pii/S1053811915003961)
+"""
+
+
+def register_tractogram(moving_tractogram, static_tractogram,
+                        only_rigid, amount_to_load, matrix_filename,
+                        verbose):
+
+    amount_to_load = max(250000, amount_to_load)
+
+    moving_streamlines = next(ichunk(moving_tractogram.streamlines,
+                                     amount_to_load))
+
+    static_streamlines = next(ichunk(static_tractogram.streamlines,
+                                     amount_to_load))
+
+    if only_rigid:
+        transformation_type = 'rigid'
+    else:
+        transformation_type = 'affine'
+
+    ret = whole_brain_slr(ArraySequence(static_streamlines),
+                          ArraySequence(moving_streamlines),
+                          x0=transformation_type,
+                          maxiter=150,
+                          verbose=verbose)
+    _, transfo, _, _ = ret
+    np.savetxt(matrix_filename, transfo)
+
+
+def _build_args_parser():
+    p = argparse.ArgumentParser(formatter_class=argparse.RawTextHelpFormatter,
+                                description=DESCRIPTION, epilog=EPILOG)
+
+    p.add_argument('moving_tractogram',
```

Comment:
If it is a trk do we must to require an reference ?

Reply:
I think this is just an old help message, there is no limitation on the filetype